qa/tasks/quiescer.py

```diff
@@ -0,0 +1,372 @@
+        Note: the duration returned by this funciton may be further split into two half-time
+              quiesces, subject to the self.split_if_longer logic"""
+        mu = (self.min_quiesce + self.max_quiesce) / 2
+        sigma = 3 * math.sqrt(self.max_quiesce - self.min_quiesce)
+        duration = round(self.rnd.gauss(mu, sigma), 1)
+        duration = max(duration, self.min_quiesce)
+        duration = min(duration, self.max_quiesce)
+        return duration
+
+    def tell_quiesce_leader(self, *args):
+        leader = None
+        rc = None
+        stdout = None
+
+        while leader is None and not self.is_stopped:
+            leader = self.fs.get_var('qdb_leader')
+            if leader is None:
+                self.logger.warn("Couldn't get quiesce db leader from the mds map")
+                self.sleep_unless_stopped(5)
+
+        while leader is not None and not self.is_stopped:
+            command = ['tell', f"mds.{leader}", 'quiesce', 'db']
+            command.extend(args)
+            self.logger.debug("Running ceph command: '%s'" % " ".join(command))
+            result = self.fs.run_ceph_cmd(args=command, check_status=False, stdout=StringIO(),
+                                          # (quiesce_timeout + guard) is a sensible cmd timeout
+                                          # for both `--quiesce --await` and `--release --await`
+                                          # It is an overkill for a query,
+                                          # but since it's just a safety net, we use it unconditionally
+                                          timeoutcmd=self.quiesce_timeout+self.QDB_CMD_TIMEOUT_GUARD)
+            rc, stdout = result.exitstatus, result.stdout.getvalue()
+            if rc == errno.ENOTTY:
+                try:
+                    resp = json.loads(stdout)
+                    leader = int(resp['leader'])
+                    self.logger.info("Retrying a quiesce db command with leader %d" % leader)
+                except Exception as e:
+                    self.logger.error("Couldn't parse ENOTTY response from an mds with error: %s\n%s" % (str(e), stdout))
+                    self.sleep_unless_stopped(5)
+            else:
+                break
+
+        return (rc, stdout)
+
+    def dump_ops_all_ranks(self):
+        result = self.fs.run_ceph_cmd(args=['tell', 'mds.*', 'ops', '--flags=locks'], check_status=False, stdout=StringIO(),
+                                        # (quiesce_timeout + guard) is a sensible cmd timeout
+                                        # for both `--quiesce --await` and `--release --await`
+                                        # It is an overkill for a query,
+                                        # but since it's just a safety net, we use it unconditionally
+                                        timeoutcmd=self.quiesce_timeout+self.QDB_CMD_TIMEOUT_GUARD)
+        stdout = result.stdout.getvalue()
+        self.logger.info("\n%s", stdout)
```

Comment:
We don't really want to dump this into the teuthology.log. Also, trying to dump the ops with `--flags=locks` often causes MDS failovers. Please mirror what test_quiesce.py is doing:

https://github.com/ceph/ceph/blob/d88fa5322df9b13a590a7691bcfbbdbf718e4469/qa/tasks/cephfs/test_quiesce.py#L107-L120

Reply:
I wanted to make use of `tell mds.*`, and here I'll have to cycle through all known MDS-es, it seems.
Also, I don't need the cache, right? Finally, this uses locks=True, so not addressing the potential failover, correct?

Reply:
> I wanted to make use of `tell mds.*`, and here I'll have to cycle through all known MDS-es, it seems.

You should do that anyway: we do not need to see the output from a standby.

> Also, I don't need the cache, right?

It is often very helpful.

> Finally, this uses locks=True, so not addressing the potential failover, correct?

The point is that `--flags=locks` can create hundreds of MB. It's too much to go in the teuthology.log and difficult to run tools like `jq` on.